open3d/geometry/TriangleMesh.cpp

```diff
@@ -135,21 +127,31 @@ TriangleMesh &TriangleMesh::operator+=(const TriangleMesh &mesh) {
   }
 
   // Added by polycam for case when there is a mixture of textures & materials in gltf
-  if (mesh.triangles_.size() == mesh.triangles_uvs_idx_.size()) {
-    triangles_uvs_idx_.resize(triangles_uvs_idx_.size() + mesh.triangles_uvs_idx_.size());
-    for (size_t i = 0; i < mesh.triangles_uvs_idx_.size(); i++) {
-      triangles_uvs_idx_[old_tri_num + i] = mesh.triangles_uvs_idx_[i].array() + add_triangles_uvs_idx_;
+  if (HasTriangleUvIndices() || mesh.HasTriangleUvIndices()) {
+    triangles_uvs_idx_.resize(new_tri_num, Eigen::Vector3i(-1, -1, -1));
```

Comment:
Do I spy Omri using -1 instead of optional??

Reply:
Yeah, yeah, yeah. I didn't create `triangles_uvs_idx_` and it's rules, I just have to use it.
`-1` in this context means no texture coordinates for that triangle. If any triangles have textures, `triangles_uvs_idx_` must be the same size as `triangles_` with `-1` for the triangles that don't have textures. It's a really kludgy way of mixing textured and untextured materials all within the same mesh.
Believe me, I wish we weren't using Open3D at all and I had cart blanche to create our own model representation where each material in the mesh had it's own vertex index array and templatized vertex coordinates/attributes array. It would have been much cleaner, run faster, required less code and had way less potential for bugs. But that would be a considerably bigger effort than the few days of Open3D refactor I did now and I doubt anybody but me has an appetite for that.